source/parser.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "../source/simulator.hpp"
+#include <map>
+#include <string>
+#include <cstdint>
+#include <yaml-cpp/node/node.h>
+
+
+namespace sim {
+
+	class YamlParser {
+	public:
+		void parseConfig(const std::string& filename, Simulator& simulator);
```

Comment:
I think its better to return simulator, not to get it as filed (you may unexpectedly pass simulator with some devices etc, what is unexcprcted)
```suggestion
		Simulator parseConfig(const std::string& filename);
```

Reply:
Now simulator is created inside the `YAMLParcer::parseConfig`